Bottom: c51fca71162992b9721c6a310d694b2bc0fe44e0
Top:    207801d367791d6ffa8cb5d8a06d1049c2fb76aa
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-05-02 15:11:19 +0100

hid/lesstif: Fix dropped crosshair artaefacts when dragging objects.

It seems that the idle_proc redraw code assumed the crosshair was
switched off upon entry, proceeding to redraw everything (and
obliterate the crosshair in the process). The final show_crosshair (1)
call only has any effect if the crosshair was notionally off before
hand.

As a quick work-around (you could probably do it with less flicker with
a little more logic), ensure the crosshair is turned off just prior to
redrawing the screen - so it will switch back on correctly afterwards.


---

diff --git a/src/hid/lesstif/main.c b/src/hid/lesstif/main.c
index cdaa7b0..3da94a9 100644
--- a/src/hid/lesstif/main.c
+++ b/src/hid/lesstif/main.c
@@ -2532,6 +2532,7 @@ idle_proc (XtPointer dummy)
       hid_expose_callback (&lesstif_hid, &region, 0);
       draw_grid ();
       lesstif_use_mask (0);
+      show_crosshair (0); /* To keep the drawn / not drawn info correct */
       XSetFunction (display, my_gc, GXcopy);
       XCopyArea (display, main_pixmap, window, my_gc, 0, 0, view_width,
 		 view_height, 0, 0);
